### DISCUSSION TITLE
Bidding system - Contradictory statements in the linked articles

### DISCUSSION THREAD

**Discussion by ThePokerDude** [lorserker/ben#97] [Q&A]
Hi,

which of the statements is true?

in 
https://greatbridgelinks.com/meet-ben-a-bridge-ai-with-a-difference/

"The system is nowhere explicitly defined, there is no rule programmed explicitly that with 15-17 HCP balanced you open 1NT for example. This makes the engine very small (only 5000 lines of code) and very flexible too. As far as I know, Ben is the only bridge AI which works like this."

in
https://intobridge.com/news/play-with-robots-on-intobridge/

What system does Ben play? 
2 over 1. Ben is coded to play relatively simple 2/1, with 15-17 NT, 5-card majors and 3-card minors. At the 2-level, Ben plays weak 2s in Spades, Hearts and Diamonds. The 2 Clubs opener is a big hand, game-forcing, unless 22-23 Balanced hand, which would rebid 2NT over 2D response.

regards
Andreas

**Comment by ThorvaldAagaard** [comment]
Both are partly correct, but also wrong in some way.

BEN is trained by feeding it with actual deals, and then based on the deals it creates a neural network,

So BEN can play any system if it is properly trained to that system. In this repository, there are models for many different systems.

But "The system is nowhere explicitly defined, there is no rule programmed explicitly that with 15-17 HCP balanced you open 1NT for example" is very right, as there are no rules about what to bid with 15-17, but instead a lot of deals where the opening is 1N, and if you look for characteristics for the hands opening 1N, you will notice, that it matches a 15-17 HCP hand.

The best system at the moment is playing SAYC but it is mostly based on some very specific training data we have created and trained BEN from.

Training is described here: https://github.com/lorserker/ben/tree/main/scripts/training

The BEN on IntoBridge is playing 2/1 based on training data created from GIB.

**Comment by ThePokerDude** [comment]
Thanks, got it. So it's trained with example biddings which follow the rules of 2/1.
After training with a subtential number of boards we assume that the NN resposible for bidding will suggest bids that follow the rules of 2/1.
So we can state that BEN plays 2/1.
That's pretty cool :)
